src/Helper/PaymentHelper.php

```diff
@@ -294,37 +355,67 @@ public function getTestMode(): bool {
   }
 
   /**
-   * Returns the Nets API payment endpoint.
+   * Returns the Nets API update endpoint.
    *
    * @return string
    *   The endpoint URL.
    */
-  public function getPaymentEndpoint(): string {
+  public function getUpdateReferenceInformationEndpoint(string $paymentId): string {
+    return $this->getTestMode()
+      ? 'https://test.api.dibspayment.eu/v1/payments/' . $paymentId . '/referenceinformation'
+      : 'https://api.dibspayment.eu/v1/payments/' . $paymentId . '/referenceinformation';
```

Comment:
Consider adding a `getBaseUrl` function (that handles “test mode”):

```suggestion
    return $this->getBaseUrl() . '/v1/payments/' . $paymentId . '/referenceinformation';
```